app/src/main/java/ch/onepass/onepass/ui/organization/OrganizationDashboardScreen.kt

```diff
@@ -285,6 +293,217 @@ private fun OrganizationSummaryCard(
+                          tint = MaterialTheme.colorScheme.onSurfaceVariant)
+                    }
+
+                if (eventsExpanded) {
+                  HorizontalDivider(
+                      color = MaterialTheme.colorScheme.outlineVariant, thickness = 1.dp)
+
+                  if (events.isEmpty()) {
+                    Text(
+                        text = "No events created yet.",
+                        style = MaterialTheme.typography.bodyMedium,
+                        color = TextSecondary,
+                        modifier = Modifier.padding(16.dp).align(Alignment.CenterHorizontally))
+                  } else {
+                    events.forEach { event ->
+                      EventCard(
+                          event = event,
+                          currentUserRole = currentUserRole,
+                          onScanTickets = { onScanTickets(event.eventId) },
+                          onEditEvent = { onEditEvent(event.eventId) })
+                    }
+                  }
+                }
+              }
+            }
+      }
+}
+
+/**
+ * Displays a single event card within the "Your events" dropdown.
+ *
+ * This card shows event details and action buttons ("Scan", "Edit") based on user roles.
+ *
+ * @param event The [Event] to display.
+ * @param currentUserRole The role of the currently logged-in user, used to determine button
+ *   visibility.
+ * @param onScanTickets Callback for the "Scan" button.
+ * @param onEditEvent Callback for the "Edit" button.
+ */
+@Composable
+private fun EventCard(
+    event: Event,
+    currentUserRole: OrganizationRole?,
+    onScanTickets: () -> Unit,
+    onEditEvent: () -> Unit
+) {
+  val canEdit =
+      currentUserRole == OrganizationRole.MEMBER || currentUserRole == OrganizationRole.OWNER
+  val canScan = currentUserRole != null // All roles can scan
+
+  Column(
+      modifier =
+          Modifier.fillMaxWidth()
+              .padding(16.dp)
+              .testTag(OrganizationDashboardTestTags.getEventCardTag(event.eventId))) {
+        Text(
+            text = event.title.uppercase(),
+            style = MaterialTheme.typography.titleMedium,
+            fontWeight = FontWeight.Bold,
+            color = MaterialTheme.colorScheme.onSurface,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis)
+
+        Spacer(modifier = Modifier.height(4.dp))
+
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.spacedBy(4.dp)) {
+              Box(
+                  modifier =
+                      Modifier.size(8.dp)
+                          .background(
+                              color =
+                                  when (event.status.name.lowercase()) {
+                                    "upcoming" -> EventDateColor
+                                    "published" -> EventDateColor
```

Comment:
This relies on string matching; if `EventStatus` enum names ever change or new statuses appear, this will silently break.
Just directly use `EventStatus.DRAFT, PUBLISHED` etc ... :

```suggestion
                                  when (event.status) {
                                    EventStatus.DRAFT -> EventDateColor
                                    EventStatus.PUBLISHED -> EventDateColor
```

Reply:
Thanks for spotting it. Fixed!